python/nwis_client/hydrotools/nwis_client/iv.py

```diff
@@ -438,157 +334,86 @@ def _get_raw_params_handler(
-        endDT: Union[
-            str,
-            datetime.datetime,
-            np.datetime64,
-            pd.Timestamp,
-            None,
-        ] = None,
-        period: Union[str, None] = None,
-        siteStatus: str = "all",
-        **params,
-    ):
-        params = self._get_raw_params_handler(
-            parameterCd, startDT, endDT, period, siteStatus, **params
+        # might want to move this to an enum in the future
+        factory_members = (
+            (
+                sites,
+                partial(
+                    split,
+                    values=sites,
+                    key="sites",
+                    split_threshold=max_sites_per_request,
+                    join_on=",",
+                ),
+            ),  # type: Union[Dict[str, str], List[Dict[str, str]]]
+            # NWIS IV API allows 1 state per api call
+            (
+                stateCd,
+                partial(split, values=stateCd, key="stateCd", split_threshold=1),
+            ),
+            # NWIS IV API allows 10 hucs per api call
+            (
+                huc,
+                partial(split, values=huc, key="huc", split_threshold=10, join_on=","),
+            ),
+            (
+                bBox,
+                lambda *args, **kwargs: [{"bBox": item} for item in _bbox_split(bBox)],
+            ),
+            # NWIS IV API allows 20 counties per api call
+            (
+                countyCd,
+                partial(
+                    split,
+                    values=countyCd,
+                    key="countyCd",
+                    split_threshold=20,
+                    join_on=",",
+                ),
+            ),
         )
 
-        # verify that state of interest is valid
-        if stateCd.upper() not in US_STATE_AND_TERRITORY_ABBREVIATIONS:
-            error_message = (
-                f"Provided state, {stateCd}, is not a valid U.S. State or "
-                "Territory abbreviation. See https://en.wikipedia.org/wiki/List_of_U.S._state_and_territory_abbreviations"
-            )
-            raise ValueError(error_message)
-
-        params["stateCd"] = stateCd
-
-        # requests GET call
-        return self._get_and_handle_response(parameters=params)
-
-    def get_raw_huc(
-        self,
-        huc: Union[List[Union[int, str]], str, int],
-        parameterCd: str = "00060",
-        startDT: Union[
-            str,
-            datetime.datetime,
-            np.datetime64,
-            pd.Timestamp,
-            None,
-        ] = None,
-        endDT: Union[
-            str,
-            datetime.datetime,
-            np.datetime64,
-            pd.Timestamp,
-            None,
-        ] = None,
-        period: Union[str, None] = None,
-        siteStatus: str = "all",
-        **params,
-    ):
-        params = self._get_raw_params_handler(
-            parameterCd, startDT, endDT, period, siteStatus, **params
-        )
+        filtered_members = [
```

Comment:
@jarq6c, it's worth noting here, in the future it would be pretty trivial to accept multiple major filters (sites, hucs, etc.). I don't know if that is desired as a feature, but I just wanted to mention it.